mso/resource_mso_tenant_policies_ipsla_monitoring_policy.go

```diff
@@ -0,0 +1,307 @@
+	return &schema.Resource{
+		Create: resourceMSOIPSLAMonitoringPolicyCreate,
+		Read:   resourceMSOIPSLAMonitoringPolicyRead,
+		Update: resourceMSOIPSLAMonitoringPolicyUpdate,
+		Delete: resourceMSOIPSLAMonitoringPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceMSOIPSLAMonitoringPolicyImport,
+		},
+
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"template_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 64),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"uuid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"sla_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"http", "tcp", "icmp", "l2ping",
+				}, false),
+			},
+			"destination_port": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(1, 65535),
+			},
+			"http_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"HTTP10", "HTTP11",
+				}, false),
+			},
+			"http_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"sla_frequency": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(1, 300),
+			},
+			"detect_multiplier": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(1, 100),
+			},
+			"request_data_size": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(1, 17512),
+			},
+			"type_of_service": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(0, 255),
+			},
+			"operation_timeout": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(0, 604800000),
+			},
+			"threshold": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(0, 604800000),
+			},
+			"ipv6_traffic_class": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(0, 255),
+			},
+		},
+		CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
+			slaType, _ := diff.GetOk("sla_type")
+			if slaType == "icmp" || slaType == "l2ping" {
+				diff.SetNew("destination_port", 0)
+			} else if slaType == "http" {
+				diff.SetNew("destination_port", 80)
+			}
+			return nil
```

Comment:
what is expected behaviour when the destination port is set to some other value and sla type if one of these pre-defined types, is it overwritten?